### PR TITLE
Enable using postgrestd

### PR DESCRIPTION
**Pull request by workingjubilee**
This lays out some target selection logic using a `PLRUST_TARGET` env var, adds a feature for selecting an appropriate tuple by default, and otherwise provides for actually using postgrestd. This is logically complete but doesn't actually include the full test and build script. I'm going to let CI/review make noises at me before I do anything further that would mangle that signal.

### CODE REVIEW

**Comment by workingjubilee** [comment]
( sorry, last commit was erroneous. )

**Comment by workingjubilee** [comment]
Link error against undefined symbols, but... only on macOS, apparently, as Ubuntu does fine? Okay then.

File: src/user_crate/target.rs [hunk 1]
```diff
@@ -0,0 +1,80 @@
+/*! PL/Rust adopts the tactic of always explicitly specifying which target to build for.
+    This prevents using the "fallback" logic of Cargo leaving builds in an unlabeled directory.
+    This is a precaution as PL/Rust is a cross-compiler.
+    so a normal build-and-test cycle may create artifacts for multiple targets.
+!*/
```
Comment (Hoverbear) [inline src/user_crate/target.rs:5]: I think the `!` is unnecessary but I ain't mad if it stays!

File: src/user_crate/target.rs [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        if t != &"" {
+            s.push('-');
+            s.push_str(t);
+        }
+    }
+    s
+}
+
+#[derive(Debug)]
+#[allow(dead_code)] // Such is the life of cfg code
```
Comment (Hoverbear) [inline src/user_crate/target.rs:50]: The rest of the code uses `thiserror` (see `src/error.rs`, would you be OK using that here as well?

**Comment by Hoverbear** [comment]
Really excited to give this a spin. Is there any parts of postgrestd or the other libraries you'd like me to review as well?

File: src/user_crate/target.rs (hunk 1 above)
Comment (workingjubilee) [reply src/user_crate/target.rs:5]: It feels nice and symmetric? :shrug:

File: src/user_crate/target.rs (hunk 2 above)
Comment (workingjubilee) [reply src/user_crate/target.rs:50]: Yes, that would be fine!

File: Cargo.toml [hunk 3]
```diff
@@ -55,3 +57,10 @@ panic = "unwind"
 opt-level = 3
 lto = "fat"
 codegen-units = 1
+
+[patch.crates-io]
+libc = { git = "https://github.com/workingjubilee/libc", branch = "postgres-os" }
+getrandom = { git = "https://github.com/workingjubilee/getrandom", branch = "postgres-os" }
+ring = { git = "https://github.com/workingjubilee/ring", branch = "postgres-os" }
```
[the remark this answers is not in the document]
Comment (workingjubilee) [reply Cargo.toml:64]: Yeah, that won't happen immediately here, but Soon.

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -128,10 +127,10 @@ jobs:
 
     - name: Install cargo-pgx
       continue-on-error: true # We might have it.
-      run: cargo install --force cargo-pgx --version $CARGO_PGX_VERSION
```
Comment (Hoverbear) [inline .github/workflows/ci.yml]: Is it possible we need to install the `cargo-pgx` off the branch?

**Comment by Hoverbear** [comment]
Hmmm so does this mean `#[no_std]` in `src/user_crate/skeleton.rs` is not required?

**Comment by workingjubilee** [comment]
Yeah, I thiiink the `#![no_std]` annotation is no longer required.

**Comment by workingjubilee** [comment]
Okay, I am throwing in the towel on making Intel Mac work. One is in the mail to me so I will be able to plug it in and poke around and fix this soon, but for right now, eff it.

**Comment by Hoverbear** [comment]
@workingjubilee I'm fine with not doing this on intel Macs yet.  They can just not use that feature. It is a dying platform anyways.

**Review by Hoverbear — APPROVED** [review]

**Comment by Hoverbear** [comment]
Both #51 and #57 work perfectly fine on top this branch, good job. :)

**Pull request merged**